secure/storage/src/tests/vault.rs

```diff
@@ -6,42 +6,174 @@ use crate::{
 
-    let mut storage0 = VaultStorage::new(
-        VAULT_HOST.into(),
-        VAULT_ROOT_TOKEN.into(),
-        Some("test0".into()),
+    // Verify creations are unique across namespaces
+    storage_1
+        .create(U64_KEY_1, Value::U64(U64_VALUE_1), &Policy::public())
+        .unwrap();
+    assert_eq!(
+        storage_1.get(U64_KEY_1).unwrap().value.u64().unwrap(),
+        U64_VALUE_1
     );
-    let mut storage1 = VaultStorage::new(
-        VAULT_HOST.into(),
-        VAULT_ROOT_TOKEN.into(),
-        Some("test1".into()),
+    assert!(storage_2.get(U64_KEY_1).is_err());
+    assert!(storage_3.get(U64_KEY_1).is_err());
+
+    // Verify create and sets are unique across namespaces
+    storage_2
+        .create(U64_KEY_2, Value::U64(U64_VALUE_2), &Policy::public())
+        .unwrap();
+    storage_2.set(U64_KEY_2, Value::U64(U64_VALUE_2)).unwrap();
+    assert_eq!(
+        storage_2.get(U64_KEY_2).unwrap().value.u64().unwrap(),
+        U64_VALUE_2
     );
+    assert!(storage_1.get(U64_KEY_2).is_err());
+    assert!(storage_3.get(U64_KEY_2).is_err());
+
+    // Verify create_if_not_exists are unique across namespaces
+    storage_3
+        .create_if_not_exists(U64_KEY_3, Value::U64(U64_VALUE_3), &Policy::public())
+        .unwrap();
+    assert_eq!(
+        storage_3.get(U64_KEY_3).unwrap().value.u64().unwrap(),
+        U64_VALUE_3
+    );
+    assert!(storage_1.get(U64_KEY_3).is_err());
+    assert!(storage_2.get(U64_KEY_3).is_err());
+}
+
+/// Runs the test suite on a VaultStorage instance that does not use distinct namespaces
+fn test_suite_no_namespaces() {
+    let mut storage = create_initialized_vault_storage(None);
+    suite::execute_all_storage_tests(&mut storage);
+}
+
+/// Runs the test suite on a VaultStorage instance that supports multiple distinct namespaces.
+/// Tests should be able to run across namespaces without interfering.
+fn test_suite_multiple_namespaces() {
+    let mut storage_1 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_1.into()));
+    let mut storage_2 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_2.into()));
+
+    suite::execute_all_storage_tests(&mut storage_1);
+    suite::execute_all_storage_tests(&mut storage_2);
+}
+
+/// Creates and initializes a VaultStorage instance for testing. If a namespace is specified, the
+/// instance will perform all storage operations under that namespace.
+fn create_initialized_vault_storage(namespace: Option<String>) -> VaultStorage {
+    let mut storage = VaultStorage::new(VAULT_HOST.into(), VAULT_ROOT_TOKEN.into(), namespace);
+    storage
+        .reset_and_clear()
+        .expect("Failed to reset VaultStorage after creation!");
 
-    test_vault(&mut storage0);
-    test_vault(&mut storage1);
-    suite::execute_all_storage_tests(&mut storage0);
-    suite::execute_all_storage_tests(&mut storage1);
-    storage.reset_and_clear().unwrap();
+    initialize_vault_storage(&mut storage);
```

Comment:
let's not call this here but rather in the test function

Reply:
Done, thanks!